CONTRIBUTING.md

```diff
@@ -137,18 +137,21 @@ If we accept the PR, a [maintainer](MAINTAINERS.md) will merge your change and u
-    1. This makes the integration frequency much higher and allows us to catch integration issues much quicker.
-    2. This still lets collaboratively develop on a big feature that is not ready for main.
-2. Treat feature branch PR's the same as PR's to `main`. CI should run on all PR's, no incomplete work should be merged, tests are necessary, etc.
-   1. This maintains the code quality going into each feature making the integration to main PR's much easier and quicker.
-   2. More visibility during development since it gives reviewers the necessary time to review each PR.
-3. Feature specific labels: This helps identify feature related issues and PR's.
+But sometimes it may be useful to create a feature branch.  This allows you work on long-running disruptive features in isolation.   The reason we don't recommend it is because it still requires maintainer access to merge changes, and the overhead of rebasing is high.  If you do want to use a feature branch:
+  
+1. Treat feature branch PR's the same as PR's to `main`. CI should run on all PR's, no incomplete work should be merged, tests are necessary, etc.
+   a. This maintains the code quality going into each feature making the integration to main PR's much easier and quicker.
+   b. More visibility during development since it gives reviewers the necessary time to review each PR.
+2. Please use Feature specific labels: This helps identify feature related issues and PR's.
 
 All the safeguard here are not rules but guidelines and should be adopted by each repository based on their specific requirements. This is to ensure that feature branch development is less likely to have code quality issues and massive merge to main PR's.
 
 For contributors looking to add a new feature that would require the creation of a feature branch, the process begins by opening an issue in the repo with the feature proposal. Depending on the nature of the feature, the maintainers of the project can decide to create a feature branch and use this model.
+
+## Experimental Features
+
+Experimental releases are used to gather early feedback on features. Features should only be marked experimental if there is a high likelihood that API or experience will change. We strongly advise people to avoid using experimental features in production as there is no guarantee the API or experience won't change before release. It is best to avoid experimental feature releases unless it is necessary. Our goal is to have all experimental features into production or removed within 2 minor releases.  We generally use [Feature Flags](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#experimental-development) to isolate experimental features in backend code.
```

Comment:
should we also call out feature flag practice for OSD and Plugins?

Reply:
Good idea!  I scanned https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/CONTRIBUTING.md and didn't see anything I can point to.  Do you want to add something and I'll point to it from here?

Reply:
I know our customers trust us and deploy the distribution to production environments. 
By definition, experimental is the opposite of stable-production.

Do we have any data points or anecdotes from customers or users asking for experimental features to be made available in a production distribution ?

If the purpose is to gather early feedback; Should we have a new experimental distribution as a separate artifact which we release ? This could also lead to more innovation and encourage experimentation by the community.

Trust is hard earned and easily lost. I would encourage us to keep a clean separation between stable-production and experimental.

my 2cs, on this above as a maintainer. Lets consider other thoughts from everyone too. thanks -Ani

Reply:
@CEHENKLE 
here is example for OSD
https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/data_source#configuration